salt/addons/init.sls

```diff
@@ -1,67 +1,39 @@
-include:
-  - kube-apiserver
-  - kubernetes-common
-
 /etc/kubernetes/addons:
   file.directory:
     - user:     root
     - group:    root
     - dir_mode: 755
     - makedirs: True
 
-{% if salt['pillar.get']('addons:dns', 'false').lower() == 'true' or salt['pillar.get']('addons:tiller', 'false').lower() == 'true' %}
 /etc/kubernetes/addons/namespace.yaml:
   file.managed:
     - source:      salt://addons/addons/namespace.yaml.jinja
     - template:    jinja
 
-create_namespace.sh:
-  cmd.script:
-    - source:      salt://addons/create_namespace.sh
+apply_namespace:
+  cmd.run:
+    - name: |
+        kubectl apply -f /etc/kubernetes/addons/namespace.yaml
+    - env:
+      - KUBECONFIG: {{ pillar['paths']['kubeconfig'] }}      
     - require:
       - service:   kube-apiserver
       - file:      /etc/kubernetes/addons/namespace.yaml
 
-kube_apiserver_ready:
-  # We need to wait for Kube API server to actually start, see k8s issue #47739
-  # TODO: Salt doesn't seem to have a retry mechanism in the version were using,
-  # so I'm doing a horrible hack right now.
-  cmd.run:
-    - name: |
-        ELAPSED=0
-        until curl --insecure --silent --fail -o /dev/null http://127.0.0.1:8080/healthz ; do
-            [ $ELAPSED -gt 300 ] && exit 1
-            sleep 1 && ELAPSED=$(( $ELAPSED + 1 ))
-        done
-        echo changed="no"
-    - stateful: True
-{% endif %}
-
 {% if salt['pillar.get']('addons:dns', 'false').lower() == 'true' %}
-/etc/kubernetes/addons/kubedns-cm.yaml:
-  file.managed:
-    - source:      salt://addons/addons/kubedns-cm.yaml.jinja
-    - template:    jinja
-
 /etc/kubernetes/addons/kubedns.yaml:
   file.managed:
     - source:      salt://addons/addons/kubedns.yaml.jinja
     - template:    jinja
 
-/etc/kubernetes/addons/kubedns-svc.yaml:
-  file.managed:
-    - source:      salt://addons/addons/kubedns-svc.yaml.jinja
-    - template:    jinja
-
-deploy_dns.sh:
-  cmd.script:
-    - source:      salt://addons/deploy_dns.sh
+apply_dns:
```

Comment:
Is it worth converting all of the existing states from '-' to '_' (or vice versa) just for convention sake? If we want to standardize on a convention going forward, I think we should put this in the README for the repo and/or the project, and then we can enforce it once folks are on the same page. But until then, please let's continue to make progress on PRs (especially for states like deploy_dns / apply_dns which already exist today).